src/services/plugin.service.ts

```diff
@@ -3,12 +3,37 @@ import { PluginLoader } from '../bootstrap/plugin.loader';
 import Constants from '../common/constants';
 import { IPlugin, ICommandLookup, IPluginLookup, IContainer } from '../common/types';
 
+export interface IPluginState {
+  name: string;
+  isActive: boolean;
+  guildID: string;
+}
+
 export class PluginService {
   public plugins: IPluginLookup = {};
   public aliases: ICommandLookup = {};
 
   private readonly _NUM_DISPLAY = 10;
 
+  public async initPluginState(container: IContainer): Promise<void> {
+    const pluginStates = (await container.storageService.getCollections()).pluginState;
+
+    if (!pluginStates) {
+      return;
+    }
+
+    const fetchedStates = await pluginStates.find().toArray();
+
+    // Set all of the plugins to the persisted state.
+    Object.values(this.plugins).forEach(plugin => {
+      fetchedStates.forEach(state => {
+        if (state.name === plugin.name && state.guildID === container.guildService.get().id) {
+          plugin.isActive = state.isActive;
+        }
```

Comment:
I would prefer to either filter the plugins with guildID or better yet, pass in the guildID for `.find()` from mongo